envoy/network/socket.h

```diff
@@ -324,6 +331,11 @@ class ConnectionInfoSetter : public ConnectionInfoProvider {
    * @param filter_chain_info the filter chain info provider backing this socket.
    */
   virtual void setFilterChainInfo(FilterChainInfoConstSharedPtr filter_chain_info) PURE;
+
+  /**
+   * @param listener_info the listener info provider backing this socket.
+   */
+  virtual void setListenerInfo(std::shared_ptr<const ListenerInfo> listener_info) PURE;
```

Comment:
nit: `ListenerInfoConstSharedPtr`.

Reply:
This is because I can't import `listener.h` due to a header cycle.